File: ios/Plugin/LoggerConfigProvider.swift
import Foundation
#if canImport(Capacitor)
import Capacitor

class LoggerConfigProvider {
    func getConfig(_ plugin: CAPPlugin) -> [String: Any] {
        guard let bridgeConfig = plugin.bridge?.config else {
            return defaults()
        }
        let pluginConfig = bridgeConfig.getPluginConfig("PicaNetworkLogger")
        let enabled = pluginConfig.getBoolean("enabled", true)
        let notify = pluginConfig.getBoolean("notify", true)
        let maxBodySize = pluginConfig.getInt("maxBodySize", 131072)
        let redactHeaders = bridgeConfig.getPluginConfigValue("PicaNetworkLogger", "redactHeaders") as? [String] ?? []
        let redactJsonFields = bridgeConfig.getPluginConfigValue("PicaNetworkLogger", "redactJsonFields") as? [String] ?? []
        return [
            "enabled": enabled,
            "notify": notify,
            "maxBodySize": maxBodySize,
            "redactHeaders": redactHeaders,
            "redactJsonFields": redactJsonFields
        ]
    }

    private func defaults() -> [String: Any] {
        return [
            "enabled": true,
            "notify": true,
            "maxBodySize": 131072,
            "redactHeaders": [] as [String],
            "redactJsonFields": [] as [String]
        ]
    }
}
#endif

File: android/src/main/java/com/linakis/capacitorpicanetworklogger/LoggerConfigProvider.kt
package com.linakis.capacitorpicanetworklogger

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin

class LoggerConfigProvider {
    fun getConfig(plugin: Plugin): JSObject {
        val config = plugin.bridge.config.getPluginConfiguration("PicaNetworkLogger")
        val output = JSObject()
        output.put("maxBodySize", config.getInt("maxBodySize", 131072))
        output.put("redactHeaders", config.getArray("redactHeaders"))
        output.put("redactJsonFields", config.getArray("redactJsonFields"))
        output.put("notify", config.getBoolean("notify", true))
        return output
    }
}
